pennylane_lightning/src/simulator/Constant.hpp

```diff
@@ -0,0 +1,258 @@
+// Copyright 2021 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @file Constant.hpp
+ * Defines all constants for statevector
+ */
+#pragma once
+
+#include "GateOperation.hpp"
+#include "KernelType.hpp"
+#include "TypeList.hpp"
+
+namespace Pennylane::Constant {
```

Comment:
Just thinking, would it be better to have a `Pennylane::Gates::Contant` namespace here (and maybe `GatesConstant.hpp` filename)? Happy either way, but I think it is clearer for what constants are available in the file with such a name.

Or, even a separate `gates` directory, with all gate-related operations?

Reply:
Thanks for the nice suggestion. I created the `gates` subdirectory and moved most of the gates-related files there.